### ISSUE TITLE
Entering garbled for the first time

### ISSUE THREAD

**Issue by chaojiwuxian** [socketio/socket.io-client-swift#1245]
The first data will occasionally be garbled, the following data is fine. How to solve this problem, the version is 15.1.0.

**Comment by nuclearace** [comment]
What version of socket.io are you using on the server? This sounds like you're using 1.0, which isn't supported.

**Comment by chaojiwuxian** [comment]
The server is currently using version 1.7.4.
Which version will be used without problems?
The problem is that occasionally the first data will be garbled, the following data is no problem.

**Comment by chaojiwuxian** [comment]
![1891571367745_ pic_hd](https://user-images.githubusercontent.com/12786026/67062743-d7d59d00-f196-11e9-9edf-f0e26d70e11a.jpg)

**Comment by chaojiwuxian** [comment]
![1891571367745_ pic_hd](https://user-images.githubusercontent.com/12786026/67062836-26833700-f197-11e9-9290-9022e98944e3.jpg)

**Comment by chaojiwuxian** [comment]
The installation is sent out on the mobile phone, why some mobile phones can not connect to the socket, and some mobile phones are used normally. My test machine has been no problem.

**Comment by nuclearace** [comment]
@chaojiwuxian socket.io 2.0 introduced a breaking change with how unicode is done. And 1.0 isn't supported anymore, so you'll need to the server to upgrade. The last version of the client was 9.x, although it won't work on current Swift without changes.

**Comment by chaojiwuxian** [comment]
At present, the old users can't receive the news of pushing the past, the new users are no problem, what is the cause?

**Comment by chaojiwuxian** [comment]
self.manager = [[SocketManager alloc]initWithSocketURL:url config:@{@"log":@NO,@"forcePolling":@NO}];     forcePolling What is the use

**Comment by nuclearace** [comment]
>At present, the old users can't receive the news of pushing the past

When you say can't receive, do you mean the socket isn't connecting or they get garbled text? Regardless, the server is using an old, unsupported version of socket.io, and needs to be upgraded.

**Comment by woodymoo** [comment]
I got same issue my ios app. It does not show chinese characters normally. 
after all, I have added @doubleEncodeUTF8:@true , but not working .. (@forcePolling:@YES, it is in my config)
and I upgraded the server side node.js extention: socket.io (1.7.4 to 2.3.0 ) , It is working! 

and I also tried to change to @forcePolling:@no, it also works !
I think upgrading the both websocket client and server are better solution.
